freestyle/shared/src/main/scala/freestyle/free.scala

```diff
@@ -4,6 +4,10 @@ import scala.annotation.{compileTimeOnly, StaticAnnotation}
 import scala.language.experimental.macros
 import scala.reflect.macros.blackbox
 
+trait FreeModuleLike {
```

Comment:
Even if the macro adds the extension to the class IntelliJ is still unable to see it so that is not that important at this point since IDEA can not see anything that the macro generates. Not just that but IDEA can't use any definitions in the macro to infer types in their precompilation parser which they use for code completion.

If the user extends it manually though the type alias is also seen as if it were part of the companion when it really is just declared in the trait. But don't worry about this now, if you agree that Algebra is a better name let's change it or we can down the road when the IDEA plugin makes some progress. 

Based on the scala IDEA gitter channel they are working a new more complete API that maybe makes the user extending anything not necessary.